src/annotations/actions/thunks.ts

```diff
@@ -32,15 +42,44 @@ export const fetchAndSetAnnotations = () => async (
 
 export const writeThenFetchAndSetAnnotations = (
   annotations: Annotation[]
-) => async (dispatch: Dispatch<AnnotationAction>): Promise<void> => {
-  await writeAnnotation(annotations)
+) => async (
+  dispatch: Dispatch<AnnotationAction | NotificationAction>
+): Promise<void> => {
+  try {
+    await writeAnnotation(annotations)
 
-  fetchAndSetAnnotations()(dispatch)
+    fetchAndSetAnnotations()(dispatch)
+  } catch (err) {
+    dispatch(notify(createAnnotationFailed(err)))
```

Comment:
i think this makes more sense one layer up, in the UI / component layer. what tends to happen is that you start out with a generic error handler that applies to all calls to this `writeThenFetchAndSetAnnotations`, and then as you run into errors in different places this function is called, you start having to tweak this 'generic' function until you have cases that are specific to each use of the function. then you conclude the reason it's so janky is because the responsibility actually lies in the UI layer and that each call of it probably represents a different use case in the UI. So let's just shortcut all that pain and remove this here, and don't catch the error and let it bubble up. Then, where this is called becomes something along the lines of (I haven't tested this so it may not be 100% correct):

```ts
const makeSingleClickHandler = () => {
  const createAnnotation = userModifiedAnnotation => {
    const {message, startTime} = userModifiedAnnotation
    try {
      dispatch(
        writeThenFetchAndSetAnnotations([
          {
            summary: message,
            startTime: new Date(startTime).getTime(),
            endTime: new Date(startTime).getTime(),
          },
        ])
      )
      event('xyplot.annotations.create_annotation.create')
    } catch (err) {
      dispatch(notify(createAnnotationFailed(err)))
      event('xyplot.annotations.create_annotation.failure')
    }
  }
```

this way, we can dispatch an error message and handle an annotation creation failure metric call in the same place, which to me seems to be a strong indicator this is the proper layer for this logic